Automattic/wp-calypso
# Fix/migrate enzyme to react testing library

### Changes proposed in this Pull Request:

This PR refactors the head component to use @testing-library/react instead of enzyme.

It also uses the opportunity to migrate chai assertions to jest along with the snapshots.

### Testing instructions:

Verify tests still pass: yarn run test-client client/components/head/test/index.jsx



Related to #63409
